account selector crashing after deleting wallet

using the account selector on the send page causes the app to crash after a wallet is deleted. Below is the log from the crash.

To Reproduce:
* Create a wallet
* Select the wallet using the account selector on the send or receive page
* Delete wallet 
* Navigate back to the send or receive page


panic: runtime error: index out of range [6] with length 6

goroutine 86 [running]:
github.com/planetdecred/godcr/ui.(*pageCommon).accountSelectorLayout(0xc004a00f00, 0x400, 0x0, 0x400, 0xf4200, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/compontents.go:472 +0x415
github.com/planetdecred/godcr/ui.(*receivePage).Layout.func1.1(0x400, 0x0, 0x400, 0xf4200, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, 0x5bf80c0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/receive_page.go:86 +0x86
gioui.org/layout.Inset.Layout(0x141800000, 0x141800000, 0x141800000, 0x141800000, 0x400, 0x0, 0x400, 0xf4200, 0x4000000040000000, 0x51c14c0, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210418151603-3b69b5ed0512/layout/layout.go:147 +0x4e5
github.com/planetdecred/godcr/ui.(*receivePage).pageSections.func1(0x440, 0x0, 0x440, 0xf4240, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, 0x5bf80c0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/receive_page.go:146 +0xb0
gioui.org/layout.Stack.Layout(0x4cd8800, 0x0, 0x0, 0x440, 0xf4240, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210418151603-3b69b5ed0512/layout/stack.go:60 +0x88a
github.com/planetdecred/godcr/ui/decredmaterial.Card.Layout.func1.1(0x0, 0x0, 0x440, 0xf4240, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, 0x5bf80c0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/decredmaterial/card.go:45 +0x260
gioui.org/layout.Inset.Layout(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x440, 0xf4240, 0x4000000040000000, 0x51c14c0, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210418151603-3b69b5ed0512/layout/layout.go:147 +0x4e5
github.com/planetdecred/godcr/ui/decredmaterial.Card.Layout.func1(0x0, 0x0, 0x440, 0xf4240, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, 0x5bf80c0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/decredmaterial/card.go:44 +0x133
gioui.org/layout.Stack.Layout(0x4d15c00, 0x0, 0x0, 0x440, 0xf4240, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210418151603-3b69b5ed0512/layout/stack.go:60 +0x88a
github.com/planetdecred/godcr/ui/decredmaterial.Card.Layout(0x0, 0x0, 0x0, 0x0, 0x41600000ffffffff, 0x4160000041600000, 0xb41600000, 0x0, 0x0, 0x440, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/decredmaterial/card.go:42 +0x19a
github.com/planetdecred/godcr/ui.(*receivePage).pageSections(0xc0008deb00, 0x0, 0x0, 0x440, 0xf4240, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/receive_page.go:144 +0x1bf
github.com/planetdecred/godcr/ui.(*receivePage).Layout.func1(0x0, 0x0, 0x440, 0xf4240, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, 0x5bf80c0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/receive_page.go:85 +0xc2
github.com/planetdecred/godcr/ui.(*receivePage).Layout.func4.1.2.1.1(0x0, 0x0, 0x440, 0xf4240, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, 0x5bf80c0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/receive_page.go:132 +0x6b
gioui.org/layout.(*List).Layout(0xc0008deb00, 0x0, 0x0, 0x440, 0xf4240, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210418151603-3b69b5ed0512/layout/list.go:111 +0x165
github.com/planetdecred/godcr/ui.(*receivePage).Layout.func4.1.2.1(0x0, 0x0, 0x440, 0x386, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, 0x5bf80c0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/receive_page.go:131 +0xc7
gioui.org/layout.Stack.Layout(0x4cd8800, 0x0, 0x0, 0x440, 0x386, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210418151603-3b69b5ed0512/layout/stack.go:60 +0x88a
github.com/planetdecred/godcr/ui/decredmaterial.Card.Layout.func1.1(0x0, 0x0, 0x440, 0x386, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, 0x5bf80c0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/decredmaterial/card.go:45 +0x260
gioui.org/layout.Inset.Layout(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x440, 0x386, 0x4000000040000000, 0x51c14c0, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210418151603-3b69b5ed0512/layout/layout.go:147 +0x4e5
github.com/planetdecred/godcr/ui/decredmaterial.Card.Layout.func1(0x0, 0x0, 0x440, 0x386, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, 0x5bf80c0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/decredmaterial/card.go:44 +0x133
gioui.org/layout.Stack.Layout(0x4d15c00, 0x0, 0x0, 0x440, 0x386, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210418151603-3b69b5ed0512/layout/stack.go:60 +0x88a
github.com/planetdecred/godcr/ui/decredmaterial.Card.Layout(0x0, 0x0, 0x0, 0x0, 0x41600000ffffffff, 0x4160000041600000, 0x4000000041600000, 0x0, 0x0, 0x440, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/decredmaterial/card.go:42 +0x19a
github.com/planetdecred/godcr/ui.(*receivePage).Layout.func4.1.2(0x0, 0x0, 0x440, 0x386, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, 0x5bf80c0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/receive_page.go:130 +0x1fa
gioui.org/layout.Flex.Layout(0x4000001, 0x0, 0x0, 0x440, 0x3d9, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210418151603-3b69b5ed0512/layout/flex.go:99 +0xd86
github.com/planetdecred/godcr/ui.(*receivePage).Layout.func4.1(0x0, 0x0, 0x440, 0x3d9, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, 0x5bf80c0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/receive_page.go:123 +0x29d
gioui.org/layout.Inset.Layout(0x141c00000, 0x141c00000, 0x141c00000, 0x141c00000, 0x0, 0x0, 0x440, 0x3d9, 0x4000000040000000, 0x51c14c0, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210418151603-3b69b5ed0512/layout/layout.go:147 +0x4e5
github.com/planetdecred/godcr/ui.pageCommon.UniformPadding(...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/page.go:505
github.com/planetdecred/godcr/ui.(*receivePage).Layout.func4(0x0, 0x0, 0x4a0, 0x439, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, 0x5bf80c0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/receive_page.go:122 +0x140
github.com/planetdecred/godcr/ui.pageCommon.Layout.func1.1.2(0x0, 0x0, 0x4a0, 0x439, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, 0x5bf80c0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/page.go:413 +0x52
gioui.org/layout.Flex.Layout(0x4000000, 0x640, 0x0, 0x640, 0x439, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210418151603-3b69b5ed0512/layout/flex.go:99 +0xd86
github.com/planetdecred/godcr/ui.pageCommon.Layout.func1.1(0x640, 0x0, 0x640, 0x439, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, 0x5bf80c0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/page.go:406 +0x250
gioui.org/layout.Flex.Layout(0x4000001, 0x640, 0xc7, 0x640, 0x4b0, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210418151603-3b69b5ed0512/layout/flex.go:99 +0xd86
github.com/planetdecred/godcr/ui.pageCommon.Layout.func1(0x640, 0xc7, 0x640, 0x4b0, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, 0x5bf80c0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/page.go:403 +0x2a2
gioui.org/layout.Stack.Layout(0x4cd7000, 0x0, 0x0, 0x640, 0x4b0, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210418151603-3b69b5ed0512/layout/stack.go:78 +0x6d7
github.com/planetdecred/godcr/ui.pageCommon.Layout(0xc00025a320, 0xc00064a540, 0xc00016e2a0, 0xc000476ff0, 0xc000476f80, 0xc000156100, 0xc00062ce10, 0xc00062ce60, 0xc00062ceb0, 0xc00062cf00, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/page.go:396 +0x55f
github.com/planetdecred/godcr/ui.(*receivePage).Layout(0xc0008deb00, 0x0, 0x0, 0x640, 0x4b0, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/receive_page.go:121 +0x2a6
github.com/planetdecred/godcr/ui.(*Window).ReceivePage.func1(0x0, 0x0, 0x640, 0x4b0, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, 0x5bf80c0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/receive_page.go:76 +0x127
gioui.org/layout.Stack.Layout(0x4cd8f01, 0x640, 0x4b0, 0x640, 0x4b0, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210418151603-3b69b5ed0512/layout/stack.go:60 +0x88a
github.com/planetdecred/godcr/ui/decredmaterial.(*Theme).Background(0xc000156100, 0x640, 0x4b0, 0x640, 0x4b0, 0x4000000040000000, 0x51c14c0, 0xc00048b090, 0xc01c15e4d09072b0, 0xbbd97dc60, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/decredmaterial/theme.go:131 +0x23d
github.com/planetdecred/godcr/ui.(*Window).Loop(0xc000476f00, 0xc000181ce0)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/window.go:237 +0x61d
created by main.main
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/main.go:128 +0x6ab

--- a/ui/send_page.go
+++ b/ui/send_page.go
@@ -1117,7 +1117,6 @@ func (pg *sendPage) Handle(c pageCommon) {
 				pg.rightExchangeValue = "USD"
 			}
 		}
-
 		pg.calculateValues(c, true)
 	}
 

--- a/ui/state.go
+++ b/ui/state.go
@@ -83,7 +83,7 @@ func (win *Window) updateStates(update interface{}) {
 		win.window.Invalidate()
 	case wallet.DeletedWallet:
 		win.selected = 0
-		win.changePage(PageWallet)
+		win.changePageAndRefresh(PageWallet)
 		win.notifyOnSuccess("Wallet removed")
 	case wallet.AddedAccount:
 		win.notifyOnSuccess("Account created")

--- a/ui/wallet_page.go
+++ b/ui/wallet_page.go
@@ -63,6 +63,7 @@ type walletPage struct {
 	watchOnlyWalletMoreButtons                 map[int]decredmaterial.IconButton
 	shadowBox                                  *decredmaterial.Shadow
 	separator                                  decredmaterial.Line
+	refreshPage                                *bool
 }
 
 func (win *Window) WalletPage(common pageCommon) layout.Widget {
@@ -82,6 +83,7 @@ func (win *Window) WalletPage(common pageCommon) layout.Widget {
 		openPopupIndex:           -1,
 		shadowBox:                common.theme.Shadow(),
 		separator:                common.theme.Separator(),
+		refreshPage:              &win.refreshPage,
 	}
 
 	pg.separator.Color = common.theme.Color.Background
@@ -273,6 +275,11 @@ func (pg *walletPage) showImportWatchOnlyWalletModal(common pageCommon) {
 
 // Layout lays out the widgets for the main wallets pg.
 func (pg *walletPage) Layout(gtx layout.Context, common pageCommon) layout.Dimensions {
+	if *pg.refreshPage {
+		common.refreshWindow()
+		*pg.refreshPage = false
+	}
+
 	if common.info.LoadedWallets == 0 {
 		return common.Layout(gtx, func(gtx C) D {
 			return common.UniformPadding(gtx, func(gtx C) D {

--- a/ui/wallet_settings_page.go
+++ b/ui/wallet_settings_page.go
@@ -176,6 +176,14 @@ func (pg *walletSettingsPage) bottomSectionLabel(title string) layout.Widget {
 	}
 }
 
+func (pg *walletSettingsPage) resetSelectedWallet(common pageCommon) {
+	common.wallAcctSelector.selectedSendWallet = 0
+	common.wallAcctSelector.selectedSendAccount = 0
+
+	common.wallAcctSelector.selectedReceiveWallet = 0
+	common.wallAcctSelector.selectedReceiveAccount = 0
+}
+
 func (pg *walletSettingsPage) handle(common pageCommon) {
 	for pg.changePass.Clicked() {
 		walletID := pg.walletInfo.Wallets[*common.selectedWallet].ID
@@ -241,7 +249,7 @@ func (pg *walletSettingsPage) handle(common pageCommon) {
 							title:    "Confirm to remove",
 							confirm: func(pass string) {
 								pg.wal.DeleteWallet(walletID, []byte(pass), pg.errorReceiver)
-								common.resetSelected()
+								pg.resetSelectedWallet(common)
 							},
 							confirmText: "Confirm",
 							cancel:      common.closeModal,

--- a/ui/wallet_settings_page.go
+++ b/ui/wallet_settings_page.go
@@ -241,6 +241,7 @@ func (pg *walletSettingsPage) handle(common pageCommon) {
 							title:    "Confirm to remove",
 							confirm: func(pass string) {
 								pg.wal.DeleteWallet(walletID, []byte(pass), pg.errorReceiver)
+								common.resetSelected()
 							},
 							confirmText: "Confirm",
 							cancel:      common.closeModal,

--- a/ui/window.go
+++ b/ui/window.go
@@ -61,9 +61,7 @@ type Window struct {
 	sysDestroyWithSync      bool
 	walletAcctMixerStatus   chan *wallet.AccountMixer
 	internalLog             chan string
-
-	selectedProposal        *dcrlibwallet.Proposal
-	refreshPage bool
+	refreshPage             bool
 }
 
 type WriteClipboard struct {

--- a/ui/window.go
+++ b/ui/window.go
@@ -61,6 +61,9 @@ type Window struct {
 	sysDestroyWithSync      bool
 	walletAcctMixerStatus   chan *wallet.AccountMixer
 	internalLog             chan string
+
+	selectedProposal        *dcrlibwallet.Proposal
+	refreshPage bool
 }
 
 type WriteClipboard struct {
@@ -122,6 +125,11 @@ func CreateWindow(wal *wallet.Wallet, decredIcons map[string]image.Image, collec
 func (win *Window) changePage(page string) {
 	win.current = page
 	win.refresh()
+}
+
+func (win *Window) changePageAndRefresh(page string) {
+	win.refreshPage = true
+	win.changePage(page)
 }
 
 func (win *Window) refresh() {

